tests/test_helpers/test_dataset_utils.py

```diff
@@ -0,0 +1,667 @@
+            "description": "An example dataset",
+            "collections": [
+                {
+                    "name": "user",
+                    "fields": [
+                        {
+                            "name": "a",
+                            "data_categories": ["user.provided"],
+                            "fidesops_meta": {"data_type": "string"},
+                        }
+                    ],
+                }
+            ],
+        }
+
+    def test_missing_collection(self):
+        """
+        Ensures that an existing collection is preserved if the API data is empty for a collection
+        """
+
+        existing_dataset = {
+            "fides_key": "example",
+            "name": "Example Dataset",
+            "description": "An example dataset",
+            "collections": [
+                {
+                    "name": "user",
+                    "fields": [
+                        {
+                            "name": "a",
+                            "data_categories": ["user.provided"],
+                            "fidesops_meta": {"data_type": "string"},
+                        }
+                    ],
+                }
+            ],
+        }
+        api_data = {}
+
+        assert generate_dataset(existing_dataset, api_data) == {
+            "fides_key": "example",
+            "name": "Example Dataset",
+            "description": "An example dataset",
+            "collections": [
+                {
+                    "name": "user",
+                    "fields": [
+                        {
+                            "name": "a",
+                            "data_categories": ["user.provided"],
+                            "fidesops_meta": {"data_type": "string"},
+                        }
+                    ],
+                }
+            ],
+        }
+
+    def test_collection_order(self):
+        """
+        Ensures that the collection matches the collection order in the existing dataset
+        """
+
+        existing_dataset = {
+            "fides_key": "example",
+            "name": "Example Dataset",
+            "description": "An example dataset",
+            "collections": [
+                {
+                    "name": "user",
+                    "fields": [
+                        {
+                            "name": "a",
+                            "data_categories": ["user.provided"],
+                            "fidesops_meta": {"data_type": "string"},
+                        }
+                    ],
+                },
+                {
+                    "name": "posts",
+                    "fields": [
+                        {
+                            "name": "b",
+                            "data_categories": ["user.provided"],
+                            "fidesops_meta": {"data_type": "string"},
+                        }
+                    ],
+                },
+            ],
+        }
+        api_data = {}
```

Comment:
would a better test here include an `api_data` object that has at least one of the existing collection elements placed in a different order than in `existing_dataset`?